nall/nall/string.hpp

```diff
@@ -39,8 +39,7 @@ struct string_view {
   string_view(const char* data);
   string_view(const char* data, u32 size);
   string_view(const string& source);
-  template<typename... P> string_view(P&&... p);
-  ~string_view();
+  string_view(string&& source) = delete;
```

Comment:
I can construct a std::string_view from a std::string&&. Can you make a case for why nall should be stricter than the C++ standard library?

I'm not convinced that forcing callers to convert a temporary into a local is really solving a problem, because if the callee holds onto a reference to the memory pointed to by the string_view it could still very well outlast the local variable in the caller's frame. I feel the onus is on the callee to read the contents of a string_view before returning.